### DISCUSSION TITLE
Status

### DISCUSSION THREAD

**Discussion by soupday** [soupday/cc3_blender_tools#2] [General]
# WIP: Version 0.2.x

**Current build may or may not be working depending on how much I mangled the code the night before.**

### Changes Completed:
 - When no texture maps are present for an advanced node group, does not generate the node group.
 - When exporting morph characters with .fbxkey or .objkey files, the key file is copied along with the export.
 - Function added to reset preferences to default values.
 - Alpha blend settings and back face culling settings can be applied to materials in the object now.
 - Option to apply alpha blend settings to whole object(s) or just active materal.
 - Remembers the applied alpha blend settings and re-applies when rebuilding materials.

### Physics System Added:
 - Uses the physX weight maps to auto-generate vertex pin weights for cloth/hair physics (Optional in the preferences).
 - Automatically sets up cloth/hair physics modifiers (Optional in the preferences)
 - Physics cloth presets can be applied to the selected object(s) and are remembered with rebuilding materials.
 - Weight maps can be added/removed to the individual materials of the objects.
 - Operator to setup and begin texture painting of the per material weight maps.

### To Do List
 - Save any modified weight maps to disk.
 - Preferences for physics settings.
 - Pick Scalp Material (Is there an eye dropper for materials?).
 - Button to auto transfer skin weights to accessories.

### FUTURE PLANS
 - At some point I'm going to have to instance the properties on a per character basis, so the add-on can deal with multiple characters at a time, maybe even deal with whole scene exports from iClone. May need significant code changes and will be messy...
 - Get all search strings to identify material type: skin, eyelashes, body, hair etc... from preferences the user can customise.
     (Might help with Daz conversions and/or 3rd party characters that have non-standard or unexpected material names.)

 - Automatically generate full FK/IK rig with Rigify or Auto-Rig Pro (Optional in the preferences) or just make my own rig up.
   (Not a clue how to do this, may take some time...)
     (Not sure if this is possible to invoke these add-ons from code...)

**Comment by greggpatton** [comment]
Hello, I appreciate your efforts on this. I think it's going to save people a lot of time.

I am interested in participating in your project if you would like some help with testing, coding, etc.  If you're interested I will be happy to pm you details about my experience via the Reallusion forum.  

This is very cool!

Thank you!

**Reply by soupday** [reply]
Testing and feedback is always welcome, in particular testing unforeseen situations such as missing or broken texture files, that sort of thing.

When the code is more stable and organised I might be more open to letting collaborators lose on it, but at the moment I'm still figuring stuff out about Python and Blender so too much of it is subject to change, I'd just be wasting your time.

**Reply by greggpatton** [reply]
Not a problem. I completely understand. Thanks!

**Comment by PhotoGranary** [comment]
Thank you for this great add-on for Blender. It is really helpful and it saves so much time. When I import CC3 characters with your add-on (high quality) into Blender and render with Cycles, I get hard shadows on the model. Do you have any idea where this could come from?
Best regards,
Simon

**Reply by PhotoGranary** [reply]
Update: I have the impression that the Subdivision Modifier does not work afterwards or has only a minor effect

**Reply by soupday** [reply]
Is it the auto smoothing? Or the custom split normals in the geometry data? Sometimes they give me trouble coming from CC3 and I have to turn them off.

**Reply by PhotoGranary** [reply]
I tried the two of them but nothing helped. But I've found out that only the Nishita sky or Pure sky add-ons cause problems with the shadows in cycles render. With simple HDRI backrounds everything is ok. However, this is also the case via the normal classic import of the model. But thanks for the tip with the geometry data.
